sdks/python/apache_beam/runners/common.py

```diff
@@ -51,117 +51,150 @@ def receive(self, windowed_value):
     raise NotImplementedError
 
 
-class DoFnRunner(Receiver):
-  """A helper class for executing ParDo operations.
+class Method(object):
+  """Represents a method of a DoFn object."""
+
+  def __init__(self, method_value, args, defaults):
+    self.args = args
+    self.defaults = defaults
+    self._method_value = method_value
+
+  def __call__(self, *args, **kwargs):
```

Comment:
This worries me from a performance standpoint. One of the key reasons to have different invokers (methods) was to avoid the generic overhead for the simple case(s).